### DISCUSSION TITLE
Question: Correct way to set .Net Maui MediaElement Source from Code?

### DISCUSSION THREAD

**Discussion by dgerding** [CommunityToolkit/Maui#1020] [General]
My question: How can I correctly instantiate a MediaSource instance and/or the MediaElement using code behind so that I can:

- Programmatically set the MediaElement Source at runtime (it's a FileSource)
- Correctly call the Play function, again from code

I'm planning to hide the control as I just want to be able to selectively play short sound "effects" without a player control visible. But first....

WelcomeVoicePlayer is a named MediaElement in the PageContent xaml. I'm trying to use that element to play a sound that I set at runtime and/or change at runtime in the ContentPage code.

Approach that's not working:
```
// Neither of these approaches to setting Source worked...
var voiceFile = FileMediaSource.FromFile(tempVoicePath);
//WelcomeVoicePlayer = new MediaElement { Source = voiceFile };
        
// When Play is called I hear nothing, even though audio file is good etc...
WelcomeVoicePlayer.Play();

```
Happy to reinstantiate the control if needed for each new source, but still need to able to set Source from code. 

Any clues appreciated.

Dave G

P.S. Big thanks to all those that brought the MediaElement forward to Maui!

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
